src/algebraic_geometry/elliptic_curve/weierstrass.lean

```diff
@@ -400,25 +400,15 @@ variables (x : R) (y : R[X])
 @[simp] noncomputable def X_class : W.coordinate_ring := adjoin_root.mk W.polynomial $ C $ X - C x
 
 lemma X_class_ne_zero [nontrivial R] : W.X_class x ≠ 0 :=
-begin
-  intro hx,
-  cases ideal.mem_span_singleton'.mp (ideal.quotient.eq_zero_iff_mem.mp hx) with p hp,
-  apply_fun degree at hp,
-  rw [W.monic_polynomial.degree_mul, degree_polynomial, degree_C $ X_sub_C_ne_zero x] at hp,
-  cases p.degree; cases hp
-end
+adjoin_root.mk_ne_zero_of_nat_degree_lt W.monic_polynomial (C_ne_zero.2 $ X_sub_C_ne_zero x) $
+  by { rw [W.nat_degree_polynomial, nat_degree_C], norm_num }
```

Comment:
I think you don't need `W.` for `nat_degree_polynomial` (untested), and if you don't need to `simp` at the end, I generally prefer `norm_num1` because it's faster.